### ISSUE TITLE
Cannot launch through Docker on Linux

### ISSUE THREAD

**Issue by maepopi** [JarodMica/ai-voice-cloning#100]
Hello!

Another Linux user coming from MRQ repo here! I'm struggling with launching the app through Docker, though there is currently no other alternative if I understand correctly. I've gone ahead and setup the docker, but when I try and launch it there's this message:

`docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: Auto-detected mode as 'legacy'
nvidia-container-cli: initialization error: load library failed: libnvidia-ml.so.1: cannot open shared object file: no such file or directory: unknown.`

Any idea as to what I should do? I'm on Ubuntu 22.04 so my nvidia driver is 535.171.04. I've seen that to upgrade it it appears I have to upgrade to Ubuntu 23 or 24 which just came out, but I'm kind of wary of doing that. Do you think there's another reason for the message I'm encountering?

Thank you very much,

Best

**Comment by maepopi** [comment]
Hey again, so I uninstalled Docker and reinstalled Docker Desktop and restarted the whole procedure but now I have this 

`2024-05-09 11:01:57 [2024-05-09 09:01:57,751] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cpu (auto detect)
2024-05-09 11:01:58 INFO:rvc.configs.config:No supported Nvidia GPU found
2024-05-09 11:01:58 INFO:rvc.configs.config:Use cpu instead
2024-05-09 11:01:58 Whisper detected
2024-05-09 11:01:58 Traceback (most recent call last):
2024-05-09 11:01:58   File "/home/user/ai-voice-cloning/./src/main.py", line 33, in <module>
2024-05-09 11:01:58     webui = setup_gradio()
2024-05-09 11:01:58             ^^^^^^^^^^^^^^
2024-05-09 11:01:58   File "/home/user/ai-voice-cloning/src/webui.py", line 600, in setup_gradio
2024-05-09 11:01:58     valle_models = get_valle_models()
2024-05-09 11:01:58                    ^^^^^^^^^^^^^^^^^^
2024-05-09 11:01:58   File "/home/user/ai-voice-cloning/src/utils.py", line 3205, in get_valle_models
2024-05-09 11:01:58     return [ f'{dir}/{d}/config.yaml' for d in os.listdir(dir) if os.path.exists(f'{dir}/{d}/config.yaml') ]
2024-05-09 11:01:58                                                ^^^^^^^^^^^^^^^
2024-05-09 11:01:58 FileNotFoundError: [Errno 2] No such file or directory: './training/'
2024-05-09 11:01:58 DEBUG:filelock:Attempting to acquire lock 140153274401488 on /home/user/.triton/autotune/Fp16Matmul_2d_kernel.pickle.lock
2024-05-09 11:01:58 DEBUG:filelock:Lock 140153274401488 acquired on /home/user/.triton/autotune/Fp16Matmul_2d_kernel.pickle.lock
2024-05-09 11:01:58 DEBUG:filelock:Attempting to release lock 140153274401488 on /home/user/.triton/autotune/Fp16Matmul_2d_kernel.pickle.lock
2024-05-09 11:01:58 DEBUG:filelock:Lock 140153274401488 released on /home/user/.triton/autotune/Fp16Matmul_2d_kernel.pickle.lock
2024-05-09 11:01:58 DEBUG:filelock:Attempting to acquire lock 140153266646224 on /home/user/.triton/autotune/Fp16Matmul_4d_kernel.pickle.lock
2024-05-09 11:01:58 DEBUG:filelock:Lock 140153266646224 acquired on /home/user/.triton/autotune/Fp16Matmul_4d_kernel.pickle.lock
2024-05-09 11:01:58 DEBUG:filelock:Attempting to release lock 140153266646224 on /home/user/.triton/autotune/Fp16Matmul_4d_kernel.pickle.lock
2024-05-09 11:01:58 DEBUG:filelock:Lock 140153266646224 released on /home/user/.triton/autotune/Fp16Matmul_4d_kernel.pickle.lock
2024-05-09 11:01:59 Press Cntrl-C to quit or application will restart... (5s)
`

I see two problems here, the lock problem and the fact that it doesn't find any GPU

**Comment by maepopi** [comment]
Okay I finally managed to get this working. Here's what I did : 

1 - Installed docker engine
Following these instructions : https://docs.docker.com/engine/install/ubuntu/#uninstall-old-versions 
- Start by uninstalling the old versions
- Then follow the "apt" repository installation
- Check that all is working using hello-world


2 - Installed nvidia docker container toolkit : https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html 
- Install with apt method
- Then run the docker configuration command
`sudo nvidia-ctk runtime configure --runtime=docker`
- restart docker

Then I re-ran setup-docker.sh, let it build, and then I ran start-docker.sh. The VM works, but now I'm stuck at another problem: the server can't be reached. The problem apparently has already been discussed here https://github.com/JarodMica/ai-voice-cloning/issues/92, but @zhenliu's solution has not worked for me. Anyway, the actual problem of launching the docker is solved so I'll close this thread
